includes/login-register.php

```diff
@@ -148,9 +148,12 @@ function edd_log_user_in( $user_id, $user_login, $user_pass, $remember = false )
 				'</a>'
```

Comment:
Note: this was done back in the previous PR, but the `wp_lostpassword_url` shouldn't necessarily presume that the redirect should go back to the checkout screen. I'm testing the profile edit screen and redirecting to checkout doesn't make sense (it's really line 147 but GH won't let me comment there).

Reply:
We could probably derive context by checking for `$_POST['action']` being `edd_process_checkout_login` -- the problem is we don't know how to find the login or profile editor pages (there is no setting). We could just go back to the homepage if we aren't on the Checkout login form.

Reply:
let's log that for a 2.10 release @robincornett